crates/core/transaction/src/gas.rs

```diff
@@ -31,9 +33,8 @@ pub trait GasCost {
 
 pub fn spend_gas_cost() -> Gas {
     Gas {
-        // Each [`Action`] has a `0` `block_space` cost, since the [`Transaction`] itself
-        // will use the encoded size of the complete transaction to calculate the block space.
-        block_space: 0,
+        // Spends are fixed size, so we hardcode the block space cost.
+        block_space: 352,
```

Comment:
Here (& elsewhere) it'd be good to have a comment with a rationale for where the 352 comes from.

Ignoring protobuf framing, I count:

192 byte proof + 64 byte auth_sig + 3 *32 byte body